evap/rewards/tests/test_tools.py

```diff
@@ -45,6 +45,20 @@ def test_semester_activated_not_all_courses(self):
         self.form.submit()
         self.assertEqual(0, reward_points_of_user(self.student))
 
+    def test_removes_from_course(self):
+        SemesterActivation.objects.create(semester=self.course.semester, is_active=True)
+        mommy.make(Course, semester=self.course.semester, participants=[self.student])
+        self.form.submit()
+        self.student.courses_participating_in.set([self.course])
+        self.assertEqual(settings.REWARD_POINTS_PER_SEMESTER, reward_points_of_user(self.student))
+
+    def test_removed_from_course(self):
```

Comment:
nice tests. however, this test class currently requires you to use that form, but that's not really the part that's tested.

so, to make clearer what's actually tested, i would suggest to make a new `class TestGrantRewardPointsParticipationChange(TestCase):` (or similar). You probably won't need a WebTest since you don't need to use an actual view for this, this also makes the tests much faster. You probably also don't need most of the stuff in the two setup-methods.

Reply:
it would be nice to test that the staff user gets an information message though.